src/components/MultiSelectInput/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import {
+    _cs,
+    listToMap,
+} from '@togglecorp/fujs';
+import { MdCheckBox, MdCheckBoxOutlineBlank } from 'react-icons/md';
+
+import SelectInputContainer from '../SelectInputContainer';
+import { rankedSearchOnList } from '../../utils';
+
+import styles from './styles.css';
+
+type OptionKey = string | number;
+interface Option {
+    [index: string]: any;
+}
+
+export interface MultiSelectInputProps<T extends OptionKey, K> {
+    value: T[],
+    onChange: (newValue: T[]) => void;
+    options: Option[],
+    keySelector: (option: Option) => OptionKey,
+    labelSelector: (option: Option) => string,
+}
+
+const Option = ({
+    children,
+    isActive,
+}) => (
+    <>
+        <div className={styles.icon}>
+            { isActive ? <MdCheckBox /> : <MdCheckBoxOutlineBlank /> }
+        </div>
+        <div className={styles.label}>
+            { children }
+        </div>
+    </>
+);
+
+function MultiSelectInput<T extends OptionKey, K>(props: MultiSelectInputProps<T, K>) {
+    const {
+        value,
+        onChange,
+        options,
+        keySelector,
+        labelSelector,
+    } = props;
+
+    const [searchInputValue, setSearchInputValue] = React.useState('');
+
+    const optionsLabelMap = React.useMemo(() => (
+        listToMap(options, keySelector, labelSelector)
+    ), [options, keySelector, labelSelector]);
+
+    const optionRendererParams = React.useCallback((key, option) => {
+        const isActive = value.indexOf(key) !== -1;
```

Comment:
It becomes a problem if no. of items in value is large.
no. of item = N
no. of totalItem = M
The total check would cost N * M.
Pessimistically speaking: N = 100, M = 10000 would result in 1000000 checks

Reply:
Should we create a valueMap as well?

Reply:
Or just add a note for now?

Reply:
Any one will do for now!

Reply:
Left a TODO